packages/edit-site/src/components/resizable-frame/index.js

```diff
@@ -0,0 +1,204 @@
+		const updatedWidth = ref.offsetWidth;
+		const updatedHeight = ref.offsetHeight;
+
+		if ( initialComputedWidthRef.current === null ) {
+			initialComputedWidthRef.current = updatedWidth;
+		}
+
+		// Set the initial aspect ratio if it hasn't been set yet.
+		if ( initialAspectRatioRef.current === null ) {
+			initialAspectRatioRef.current = updatedWidth / updatedHeight;
+		}
+
+		// Calculate the intermediate aspect ratio based on the current width.
+		const lerpFactor =
+			1 -
+			Math.max(
+				0,
+				Math.min(
+					1,
+					( updatedWidth - MIN_FRAME_SIZE ) /
+						( 1300 - MIN_FRAME_SIZE )
+				)
+			);
+
+		// Calculate the height based on the intermediate aspect ratio
+		// ensuring the frame arrives at a 9:19.5 aspect ratio.
+		const intermediateAspectRatio = lerp(
+			initialAspectRatioRef.current,
+			9 / 19.5,
+			lerpFactor
+		);
+		const newHeight = updatedWidth / intermediateAspectRatio;
+
+		setIsResizing( true );
+
+		if ( event.clientX < 344 ) {
+			const oversizeWidth = Math.max(
+				initialComputedWidthRef.current +
+					( updatedWidth - initialComputedWidthRef.current ) / 2,
+				initialComputedWidthRef.current
+			);
+
+			setFrameSize( {
+				width: oversizeWidth,
+				height: '100%',
+			} );
+			setIsOversized( true );
+		} else {
+			setFrameSize( { width: updatedWidth, height: newHeight } );
+			setIsOversized( false );
+		}
+	};
+
+	// Make the frame full screen when the user resizes it to the left.
+	const handleResizeStop = ( event ) => {
+		// Reset the initial aspect ratio if the frame is resized slightly
+		// above the sidebar but not far enough to trigger full screen.
+		if ( event.clientX < 344 && event.clientX > 200 ) {
+			setFrameSize( { width: '100%', height: '100%' } );
+		}
+
+		// Trigger full screen if the frame is resized far enough to the left.
+		if ( event.clientX < 200 ) {
+			setCanvasMode( 'edit' );
+			// Wait for the frame to animate to full screen before resetting its size.
+			const timeoutId = setTimeout( () => {
+				setFrameSize( { width: '100%', height: '100%' } );
+			}, 500 );
+
+			// Clean up the timeout when the effect is no longer needed.
+			return () => clearTimeout( timeoutId );
+		}
+
+		setIsResizing( false );
+	};
+
+	return (
+		<ResizableBox
+			as={ motion.div }
+			initial={ false }
+			animate={ {
+				flexGrow: isFull ? 1 : 0,
+				height: frameSize.height,
+			} }
+			transition={ FRAME_TRANSITION }
+			size={ frameSize }
+			enable={ {
+				top: false,
+				right: false,
+				bottom: false,
+				left: true,
+				topRight: false,
+				bottomRight: false,
+				bottomLeft: false,
+				topLeft: false,
+			} }
+			resizeRatio={ isOversized ? 1 : 2 }
+			handleClasses={ undefined }
+			handleStyles={ {
+				left: HANDLE_STYLES_OVERRIDE,
+				right: HANDLE_STYLES_OVERRIDE,
+			} }
+			minWidth={ MIN_FRAME_SIZE }
+			maxWidth={ '150%' }
+			maxHeight={ '100%' }
+			onMouseOver={ () => setIsHovering( true ) }
+			onMouseOut={ () => setIsHovering( false ) }
```

Comment:
This seems to be fine in my testing. The reason that dragging could get "stuck" was because mouseUp events weren't being captured properly, which I've [addressed in CSS](https://github.com/WordPress/gutenberg/pull/49910/commits/c685c5c28510cf1743313b4a291d0bfbd9e93dc1#diff-1f3d49a30cfabf8cad4caabd43bddcbc50ec8573e453981f9cab30433c7b1df3R12-R18).